families/supplychain/python/sawtooth_supplychain/rest_api/route_handlers.py

```diff
@@ -0,0 +1,756 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------------
+
+# Supply Chain REST API
+
+import logging
+import json
+import re
+from concurrent.futures import ThreadPoolExecutor
+import psycopg2
+import aiopg
+from aiohttp import web
+
+import sawtooth_supplychain.rest_api.exceptions as errors
+
+
+DEFAULT_TIMEOUT = 300
+LOGGER = logging.getLogger(__name__)
+DEFAULT_PAGE_SIZE = 1000    # Reasonable default
+
+
+async def db_query(db_cnx, query, query_tuple):
```

Comment:
What is a `query_tuple`? Why do I need to set the query twice, once as a string, and once as a tuple? The name is not very meaningful, and possibly confusing. Something like `query_variables` would probably be more meaningful.